sdv/metadata/table.py

```diff
@@ -643,10 +629,6 @@ def transform(self, data, on_missing_column='error'):
         Returns:
             pandas.DataFrame:
                 Transformed data.
-
-        Raises:
-            ConstraintsNotMetError:
-                If the table data is not valid for the provided constraints.
```

Comment:
Should this be replaced with the new error that gets raised?

Reply:
The `self._validate_data_on_constraints(data)` was moved from here to the `transform` function, so the error is not really from here anymore.

Reply:
Does it still surface here though?

Reply:
It's possible, I never did test whether errors that didn't happen in the first transform could be happen in the second. Regardless, I don't think we usually write the errors of children methods in the parent one, right? If so, then we need to add the error from `_validate_constraint_columns` as well.